crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -316,6 +317,13 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
         if status_reporter.errors_found() {
             errors_found = true;
         }
+        
+        match status_reporter.finalize() {
```

Comment:
still not thrilled with `finalize`, both the role and the name.  That is very much a C#-ism.